Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+    {
+        var maxHealth = component.CoreBlobTotalHealth;
+        var currentHealth = maxHealth - args.Damageable.TotalDamage;
+
+        if (component.Observer != null)
+            _alerts.ShowAlert(component.Observer.Value, AlertType.BlobHealth, (short) Math.Clamp(Math.Round(currentHealth.Float() / 10f), 0, 20));
+    }
+
+    private void OnStartup(EntityUid uid, BlobCoreComponent component, ComponentStartup args)
+    {
+        ChangeBlobPoint(uid, 0, component);
+
+        if (TryComp<BlobTileComponent>(uid, out var blobTileComponent))
+        {
+            blobTileComponent.Core = uid;
+            blobTileComponent.Color = component.ChemСolors[component.CurrentChem];
+            Dirty(blobTileComponent);
+        }
+
+        component.BlobTiles.Add(uid);
+
+        ChangeChem(uid, component.DefaultChem, component);
+    }
+
+    public void ChangeChem(EntityUid uid, BlobChemType newChem, BlobCoreComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        if (newChem == component.CurrentChem)
+            return;
+
+        var oldChem = component.CurrentChem;
+        component.CurrentChem = newChem;
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+
+            blobTileComponent.Color = component.ChemСolors[newChem];
+            Dirty(blobTileComponent);
+
+            if (TryComp<BlobFactoryComponent>(blobTile, out var blobFactoryComponent))
+            {
+                if (TryComp<BlobbernautComponent>(blobFactoryComponent.Blobbernaut, out var blobbernautComponent))
+                {
+                    blobbernautComponent.Color = component.ChemСolors[newChem];
+                    Dirty(blobbernautComponent);
+
+                    if (TryComp<MeleeWeaponComponent>(blobFactoryComponent.Blobbernaut, out var meleeWeaponComponent))
+                    {
+                        var blobbernautDamage = new DamageSpecifier();
+                        foreach (var keyValuePair in component.ChemDamageDict[component.CurrentChem].DamageDict)
+                        {
+                            blobbernautDamage.DamageDict.Add(keyValuePair.Key, keyValuePair.Value * 0.8f);
+                        }
+                        meleeWeaponComponent.Damage = blobbernautDamage;
+                    }
+
+                    ChangeBlobEntChem(blobFactoryComponent.Blobbernaut.Value, oldChem, newChem);
+                }
+
+                foreach (var compBlobPod in blobFactoryComponent.BlobPods)
+                {
+                    if (TryComp<SmokeOnTriggerComponent>(compBlobPod, out var smokeOnTriggerComponent))
+                    {
+                        smokeOnTriggerComponent.SmokeColor = component.ChemСolors[newChem];
+                    }
+                }
+            }
```

Comment:
Blob core has no business handling all this heavily duplicated code, the respective systems for each thing (factories, blobbernauts, pods, etc) should be responsible for their own updating. Fire an event.